ethcore/builtin/src/lib.rs

```diff
@@ -16,51 +16,38 @@
 
 //! Standard built-in contracts.
 
+#![warn(missing_docs)]
+
 use std::{
 	cmp::{max, min},
+	collections::BTreeMap,
+	convert::{TryFrom, TryInto},
 	io::{self, Read, Cursor},
 	mem::size_of,
+	str::FromStr
 };
 
-use bn;
-use byteorder::{BigEndian, LittleEndian, ByteOrder, ReadBytesExt};
+use byteorder::{BigEndian, LittleEndian, ReadBytesExt};
 use ethereum_types::{H256, U256};
 use ethjson;
 use ethkey::{Signature, recover as ec_recover};
-use hash::keccak;
+use eip_152::compress;
+use keccak_hash::keccak;
 use log::{warn, trace};
 use num::{BigUint, Zero, One};
-use bytes::BytesRef;
+use parity_bytes::BytesRef;
 use parity_crypto::digest;
-use eip_152::compress;
-
-/// Execution error.
-#[derive(Debug, PartialEq)]
-pub struct Error(pub &'static str);
-
-impl From<&'static str> for Error {
-	fn from(val: &'static str) -> Self {
-		Error(val)
-	}
-}
-
-impl Into<::vm::Error> for Error {
-	fn into(self) -> ::vm::Error {
-		::vm::Error::BuiltIn(self.0)
-	}
-}
 
 /// Native implementation of a built-in contract.
-pub trait Impl: Send + Sync {
+trait Implementation: Send + Sync {
 	/// execute this built-in on the given input, writing to the given output.
-	fn execute(&self, input: &[u8], output: &mut BytesRef) -> Result<(), Error>;
+	fn execute(&self, input: &[u8], output: &mut BytesRef) -> Result<(), &'static str>;
```

Comment:
I had  to replace `EthcoreError` with `&str` because importing `ethcore` doesn't work, cyclic depedency